### PR TITLE
Adding unstable documentation

### PR DESCRIPTION
**Pull request by gauravano**
Adding some words about `unstable` branch and workflow
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in a uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by gauravano** [comment]
Hi, as I mentioned in the call yesterday that I would be adding documentation for `unstable` so here it is. @rarrunategu1 want to get your review especially as you are currently trying your way with `unstable` and so please let me know, if more clarity is required. 

@jywarren am I right about the environment and other specs?  let me know, if I am missing something. Thanks!!

**Comment by sashadev-sky** [comment]
@gauravano this is an awesome idea!! Reviewing now. Lets get this into mapknitter README too :)

File: doc/TESTING.md [hunk 1]
```diff
@@ -152,3 +152,24 @@ Take a look at this test [search_service_full_text_search_test.rb](https://githu
...
+
+### Pushing on `unstable`
+
+1. Drop a message in the [chatroom](https://gitter.im/publiclab/publiclab) that you are pushing to _unstable_, as other members also use it for testing their work so you don't want to shock them by pushing your work over their. After dropping the message, wait for 5 mins, and if no one mention that they are using the branch, then go to step 2.
+2. Push using: `git push -f https://github.com/publiclab/plots2.git HEAD:unstable`.<br><br>**Note:** If you get this message - `remote: Permission to publiclab/plots2.git denied to <username>` then you don't have permission to push to _unstable_ branch.
+But, don't worry, feel free to ask in https://gitter.im/publiclab/publiclab and someone else with push permission to unstable will push your work(PR/branch) for you.  
+3. See build Progress at https://jenkins.laboratoriopublico.org/job/Plots-Unstable/. It takes 8-15 mins for build to complete. 
+4. Test website: https://unstable.publiclab.org/. 
+
+Also, if your PR is merged into plots2/master, then your changes will get deployed to https://stable.publiclab.org, so feel free to test your work there if your PR is merged.
```
Comment (sashadev-sky) [inline doc/TESTING.md]:
```suggestion
Once your PR is merged into plots2/master, code is only deployed to the live site [publiclab.org](publiclab.org) after some time by our maintainers. (They often send a message in our Gitter chatroom to notify contributors of this update). Fortunately, you can see your changes implemented immediately at https://stable.publiclab.org, which builds from the master branch. This is useful for testing your work again after a merge.
```

File: doc/TESTING.md [hunk 2]
```diff
@@ -152,3 +152,24 @@ Take a look at this test [search_service_full_text_search_test.rb](https://githu
...
+_unstable_ is one test server, which can be used by any contributor with special permission for testing their work(PRs). One needs to push their commits to _unstable_ [branch](https://github.com/publiclab/plots2/tree/unstable) and they will be deployed at https://unstable.publiclab.org/ after building. 
+
+**Environment:** Test environment
+**Database:** MySQL
+**Data:** Data present at https://unstable.publiclab.org is updated occasionally using the Production database, so it has good amount of data for testing. 
+**Email testing:** _unstable_ server can't send emails so Email related features can't be tested.  
+
+### Pushing on `unstable`
+
+1. Drop a message in the [chatroom](https://gitter.im/publiclab/publiclab) that you are pushing to _unstable_, as other members also use it for testing their work so you don't want to shock them by pushing your work over their. After dropping the message, wait for 5 mins, and if no one mention that they are using the branch, then go to step 2.
```
Comment (sashadev-sky) [inline doc/TESTING.md]:
```suggestion
1. Drop a message in the [chatroom](https://gitter.im/publiclab/publiclab) that you are pushing to _unstable_, as other members also use it for testing their work so you don't want to shock them by pushing your work over theirs. After dropping the message, wait for 5 mins, and if no one mention that they are using the branch, then go to step 2.
```

File: doc/TESTING.md [hunk 3]
```diff
@@ -152,3 +152,24 @@ Take a look at this test [search_service_full_text_search_test.rb](https://githu
 
 This way we don't have errors either using SQLite3 or MySQL on development and tests
 environments.
+
+## Testing your work with Unstable
+
+### What is _unstable_? 
+
+_unstable_ is one test server, which can be used by any contributor with special permission for testing their work(PRs). One needs to push their commits to _unstable_ [branch](https://github.com/publiclab/plots2/tree/unstable) and they will be deployed at https://unstable.publiclab.org/ after building.
```
[the remark this answers is not in the document]
Comment (gauravano) [reply doc/TESTING.md]: Hmm, I was kind of thinking that would that be good if we can start the intro as a definition? I will reframe a little to add your part too.

> Use it when you need database records that are too difficult or time-consuming to reproduce locally, or to debug a problem on the production site that you can't reproduce locally.
See summary and usage instructions below.

I will add this.

Thanks @sashadev-sky for reviewing!!

[the remark this answers is not in the document]
Comment (gauravano) [reply doc/TESTING.md]: incorporated this one. Please review again. @sashadev-sky

[the remark this answers is not in the document]
Comment (sashadev-sky) [reply doc/TESTING.md]: Sounds good!!

File: doc/TESTING.md [hunk 4]
```diff
@@ -152,3 +152,26 @@ Take a look at this test [search_service_full_text_search_test.rb](https://githu
...
+**Environment:** Test environment
+**Database:** MySQL
+**Data:** Data present at https://unstable.publiclab.org is updated occasionally using the Production database, so it has good amount of data for testing. 
+**Email testing:** _unstable_ server can't send emails so Email related features can't be tested.  
+
+### Pushing on `unstable`
+
+1. Drop a message in the [chatroom](https://gitter.im/publiclab/publiclab) that you are pushing to _unstable_, as other members also use it for testing their work so you don't want to shock them by pushing your work over theirs. After dropping the message, wait for 5 mins, and if no one mention that they are using the branch, then go to step 2.
+2. Push using: `git push -f https://github.com/publiclab/plots2.git HEAD:unstable`.<br><br>**Note:** If you get this message - `remote: Permission to publiclab/plots2.git denied to <username>` then you don't have permission to push to _unstable_ branch.
+But, don't worry, feel free to ping @publiclab/maintainers on your open PR to request access or ask in https://gitter.im/publiclab/publiclab and someone else with write access will push your work (PR/branch) for you.
```
Comment (sashadev-sky) [inline doc/TESTING.md]:
```suggestion
But, don't worry, feel free to ping @publiclab/maintainers on your open PR to request access or ask in the chatroom and someone else with write access will push your work (PR/branch) for you.  
```

File: doc/TESTING.md [hunk 5]
```diff
@@ -152,3 +152,26 @@ Take a look at this test [search_service_full_text_search_test.rb](https://githu
...
+See summary and usage instructions below.
+
+**Environment:** Test environment
+**Database:** MySQL
+**Data:** Data present at https://unstable.publiclab.org is updated occasionally using the Production database, so it has good amount of data for testing. 
+**Email testing:** _unstable_ server can't send emails so Email related features can't be tested.  
+
+### Pushing on `unstable`
+
+1. Drop a message in the [chatroom](https://gitter.im/publiclab/publiclab) that you are pushing to _unstable_, as other members also use it for testing their work so you don't want to shock them by pushing your work over theirs. After dropping the message, wait for 5 mins, and if no one mention that they are using the branch, then go to step 2.
```
Comment (sashadev-sky) [inline doc/TESTING.md]:
```suggestion
1. Drop a message in the [Gitter chatroom](https://gitter.im/publiclab/publiclab) that you are pushing to _unstable_, as other members also use it for testing their work so you don't want to shock them by pushing your work over theirs. After dropping the message, wait for 5 mins, and if no one mention that they are using the branch, then go to step 2.
```

File: doc/TESTING.md [hunk 6]
```diff
@@ -152,3 +152,26 @@ Take a look at this test [search_service_full_text_search_test.rb](https://githu
...
+2. Push using: `git push -f https://github.com/publiclab/plots2.git HEAD:unstable`.<br><br>**Note:** If you get this message - `remote: Permission to publiclab/plots2.git denied to <username>` then you don't have permission to push to _unstable_ branch.
+But, don't worry, feel free to ping @publiclab/maintainers on your open PR to request access or ask in https://gitter.im/publiclab/publiclab and someone else with write access will push your work (PR/branch) for you.  
+3. See build Progress at https://jenkins.laboratoriopublico.org/job/Plots-Unstable/. It takes 8-15 mins for build to complete. 
+4. Test website: https://unstable.publiclab.org/. 
+
+Once your PR is merged into plots2/master, code is only deployed to the live site [publiclab.org](publiclab.org) after some time by our maintainers. (They often send a message in our Gitter chatroom to notify contributors of this update). Fortunately, you can see your changes implemented immediately at https://stable.publiclab.org, which builds from the master branch. This is useful for testing your work again after a merge.
```
Comment (sashadev-sky) [inline doc/TESTING.md]:
```suggestion
Once your PR is merged into plots2/master, code is only deployed to the live site [publiclab.org](publiclab.org) after some time by our maintainers. (They often send a message in our chatroom to notify contributors of this update). Fortunately, you can see your changes implemented immediately at https://stable.publiclab.org, which builds from the master branch. This is useful for testing your work again after a merge.
```

**Comment by gauravano** [comment]
Hey @sashadev-sky, merged your suggestions. (I ignored the chatroom names due to laziness :see_no_evil: . Will be more attention next time :)

Hi @rarrunategu1, I asked you to pass PR link as I assumed that you had a PR against the master. We can also push to unstable using the remote branch of your repo. 
In case, you'll create PR against master or pass me the branch name, I can push your new commits without merging your PR as that PR would be against master.

We encourage everyone to open PR regardless of how much they have made toward solving the issue as by that other contributors can also help you and you can have discussion over the issues you face.

**Comment by gauravano** [comment]
And, you'll get the unstable permission soon. I guess @jywarren is busy right now.

**Comment by gauravano** [comment]
Hey @jywarren @sashadev-sky, anything you want to add? Thanks!

**Comment by gauravano** [comment]
Merged :+1:

**Pull request merged**